### PR TITLE
fix for RunAsNonRoot check

### PR DESCRIPTION
**Pull request by edcdavid**
In the current test, the check we check RunAsNonRoot separately for pods then for its containers. However the pod RunAsNonRoot is just a default value for containers so it should not be checked separately. The pod value for RunAsNonRoot is used only if the RunAsNonRoot is not present in the container (nil). 
This PR changes the logic of the test to use the pod configuration as a default value for the container.

see https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container

> Set the security context for a Container
> To specify security settings for a Container, include the securityContext field in the Container manifest. The securityContext field is a [SecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#securitycontext-v1-core) object. Security settings that you specify for a Container apply only to the individual Container, and they override settings made at the Pod level when there is overlap. Container settings do not affect the Pod's Volumes.

### CODE REVIEW

File: tests/accesscontrol/suite.go [hunk 1]
```diff
@@ -396,33 +396,19 @@ func testSecConRunAsNonRoot(check *checksdb.Check, env *provider.TestEnvironment
...
-				check.LogInfo("Container %q in Pod %q is running as non-root", cut.Name, put.Name)
-				compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(cut.Namespace, cut.Name, "Container is running as non-root", true))
-			} else {
-				check.LogError("Container %q in Pod %q is running as root", cut.Name, put.Name)
-				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Container is running as root", false))
+			check.LogError("Pod %q might be running as root", put.Name)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod might be running as root", false))
+			for _, cut := range nonCompliantContainers {
+				check.LogError("Container %q in Pod %q might be running as root", cut.Name, put.Name)
+				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewContainerReportObject(put.Namespace, put.Name, cut.Name, "Container might be running as root", false))
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply tests/accesscontrol/suite.go:408]: do you have a suggestion in mind? Is "might" the issue?

[the remark this answers is not in the document]
Comment (sebrandon1) [reply tests/accesscontrol/suite.go:408]: Yeah it seems too loose for a non-compliant object.

[the remark this answers is not in the document]
Comment (edcdavid) [reply tests/accesscontrol/suite.go]: Fixed

**Review by sebrandon1 — APPROVED** [review]

File: tests/accesscontrol/suite.go [hunk 2]
```diff
@@ -396,33 +396,19 @@ func testSecConRunAsNonRoot(check *checksdb.Check, env *provider.TestEnvironment
...
-			}
-
-			if cut.IsContainerRunAsNonRoot() {
-				check.LogInfo("Container %q in Pod %q is running as non-root", cut.Name, put.Name)
-				compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(cut.Namespace, cut.Name, "Container is running as non-root", true))
-			} else {
-				check.LogError("Container %q in Pod %q is running as root", cut.Name, put.Name)
-				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Container is running as root", false))
+			check.LogError("Pod %q is configured with RunAsNonRoot SCC parameter set to false for some of its containers", put.Name)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod is configured with RunAsNonRoot SCC parameter set to false for some of its containers", false))
```
Comment (greyerof) [inline tests/accesscontrol/suite.go:405]: Suggestion: imho, this reason and the previous log trace are not very accurate. In order to help users to debug this failing tc, I suggest adding the specific reason, which, if I'm not wrong, should be one of these three:
a) Neither of both container and pod's scc.runAsNonRoot fields were set (both are nil).
b) Container's scc.runAsNonRoot is not set (nil) but pod's scc.runAsNonRoot is explicitly set to false.
c) Container's scc.RunAsNonRoot was explicitly set to false.

**Review by greyerof — APPROVED** [review]

File: tests/accesscontrol/suite.go (hunk 2 above)
Comment (edcdavid) [reply tests/accesscontrol/suite.go:405]: Fixed

**Review by greyerof — APPROVED** [review]

**Pull request merged**